### ISSUE TITLE
Fresh Install - Not working correctly (in latest OSMC versions)

### ISSUE THREAD

**Issue by theboyrossy** [alombarte/raspberry-osmc-automated#9]
Hi there, I created an OSMC setup running flexget about two years ago for a friend and recently it all went wrong and needed a fresh start.

I have started from scratch and installed OSMC, all I've done in OSMC is to change the max_usb current to help with powering the external drive and set up the WIFi dongle.

I've installed flexget using your installer and come across the same errors as a previous user who post was titled "pkg_resources.DistributionNotFound: funcsigs". After following the advice in the post I have got flexget working.....mostly

The issue I have is the last step where the files are moved to the correct location in the TV Series folder, I only have one show in my rss feed currently while testing to keep things simple and I keep getting tvdb_series_name errors. (I've also had these error's with an RSS feed with many shows included) I've been struggling with this for the past few days and have yet to find a solution that works. I've tried various different config files from the cookbook but I can't seem to get things to work. 

So basically my question is why does your config that comes with the install keep failing?

`2016-11-04` 11:25 WARNING  plugin        sort-files      Path value replacement `/home/osmc/TV Shows/{{ tvdb_series_name }}` failed: (UndefinedError) 'tvdb_series_n                                      ame' is undefined
2016-11-04 11:25 ERROR    exec          sort-files      Could not set exec command for arrow.504.hdtv-lol: (UndefinedError) 'tvdb_series_name' is undefined
2016-11-04 11:25 ERROR    exec          sort-files      Could not set exec command for arrow.505.hdtv-lol: (UndefinedError) 'tvdb_series_name' is undefined
2016-11-04 11:25 INFO     manager                       Removed test `database`

**Comment by alessandro-dallatorre** [comment]
Hi, I have exactly the same issue. The move part of the config.yml script fails giving (UndefinedError) 'tvdb_series_name' is undefined. That is annoying since the downloaded files came from a know series and suddenly are unable to land to my target library folder. I have to move them and rename them by hand so far. I look all around for similar issue reports, but no solution found yet.

**Comment by alombarte** [comment]
There are different things here:
1) The installation might fail in newer OSMC versions, and there are a lot of dependencies to take into account. I haven't tried the new versions for the past year, so errors like "pkg_resources.DistributionNotFound: funcsigs" might happen and I am not aware of them until I reinstall my raspberry (which has been working flawlesly for very long and I don't feel like reinstalling it)

2) The 'tvdb_series_name' is undefined error is NOT related with the scripts, but with the services that tries to find out the Series Name based on a weird filename. Please see the first issue https://github.com/alombarte/raspberry-osmc-automated/issues/1 which is exactly this and you'll find how to fix it.

**Comment by alombarte** [comment]
Ah, and of course you can change all the `{{ tvdb_series_name }}` appearances in the settings with `{{ tvdb_series_name|default('Unknown') }}` and the script won't fail anymore. But since the service isn't able to guess the series name the file will got to a folder named "Unknown". You can do the same with all the other variables.

In my current raspberry setup and because I've tested over 50 different series so far, I've opted to not rename the filenames and only use the series name variable (with fixes in #1). The files are "ugly" but Kodi will put the metadata, covers and everything, so I don't care about the internal structure.

**Comment by theboyrossy** [comment]
Thanks for the information above, I'll get on with trying that out when I get the chance tonight.

**Comment by theboyrossy** [comment]
Unfortunately the above fixes haven't worked for me. I spent a good few hours trying to get things working last. Goth with no luck. Pulling my hair out now, none of the cookbook scripts seem to work, something must have changed.

**Comment by alombarte** [comment]
If you put your script and error logs in a gist or something I can try to see if there is something obvious to help out. It's all I can offer for the price :P

**Comment by bernardomaciel** [comment]
I am having the same issue.
I tried the `{{ tvdb_series_name|default('Unknown') }}` approach but this will not work with multiple shows that are not recognized properly. The generated filename will be the same and you will only have one file, overwritten always.

Is it possible to change `config.yml` so that when any of the variables substitutions fails (for `tvdb_series_name`, `series_id` and `tvdb_ep_name`), the file is moved to `/home/osmc/TV Shows/Unknown` without renaming?

For me, that would suffice.

**Comment by alombarte** [comment]
@bernardomaciel Yes, this is possible.

The solution is simple, remove or comment the `filename` and that would be all. The section `move` will move the file to the path specified as `to:` and since no `filename` will be set the original filename will be kept.

Most of the problems are solved by declaring the series. This is an example that I just reproduced:

```
2016-11-13 15:56 ERROR    entry         sort-files      Failed shameless.us.706.hdtv-lol (Path value replacement `/home/osmc/TV Shows/{{ tvdb_series_name }}` failed: (UndefinedError) 'tvdb_series_name' is undefined)
```

Then I edit the config file and I add:

```
series:
  - shameless
```

And when I run again:

```
2016-11-13 16:00 VERBOSE  task          sort-files      ACCEPTED: `shameless.us.706.hdtv-lol` by accept_all plugin
```

It works. There are other more complex cases where you need to play with the `tvdb_id` plugin or add a `manipulate` section, like this:

```
series:
      - greys.anatomy:
          set:
            tvdb_id: 73762
            ep_regexp: (\d\d)(\d\d)     # must return TWO groups, episode and chapter.
```

(the show id can be searched in http://thetvdb.com/),
or

```
 manipulate:
                - title:
                    replace:
                      regexp: 'designated.survivor.(\d)(\d\d)'
                      format: 'designated.survivor.S\1E\2'
```

But I think that the effort is not worth it, and in my setup I don't rename the files at all. The Kodi is able to find the metadata easily anyway.

I would also suggest to have a look at the plugin that http://flexget.com/Plugins/thetvdb_lookup

**Comment by bernardomaciel** [comment]
Thanks, that works 👍 
I also changed the `exec` part for subliminal not to fail:

```
    # TVDB doesn't recognise "Adventure Time with Finn and Jake" so you must add such exceptions here manually
    series:
      - Adventure Time
    move:
      # this is where the series will be put.
      to: /home/osmc/TV Shows/
    # Download subtitles of the Series in your chosen language:
    exec: /usr/local/bin/subliminal download -l "en" -s "/home/osmc/TV Shows/"
```

I specified the directory; is there a way to reference only the file(s) that should be processed?

**EDIT** You commented that

> But I think that the effort is not worth it, and in my setup I don't rename the files at all. The Kodi is able to find the metadata easily anyway.

How does your Kodi setup find that metadata? If I leave all files (not renamed) in a single directory, the TvDB scraper fails all look ups!
Ref.: [kodi - How can I get media files to display in a library style? - Raspberry Pi Stack Exchange](http://raspberrypi.stackexchange.com/a/46872)

**Comment by theboyrossy** [comment]
"If you put your script and error logs in a gist or something I can try to see if there is something obvious to help out."

I'm afraid I don't know how to do that, but the script is literally the script that was supplied by the installer but with one series defined 'Arrow' with its tvdb ID set. My showrss feed has a single show in it and I have commented out the subtitles section.

**Comment by alombarte** [comment]
@bernardomaciel **I do keep** the TV Show name folder. What I don't do is renaming the filename based on Thetvdb data.

@theboyrossy You only need to paste your configuration file here: https://gist.github.com/ and share the link. Remember also to share the whole flexget execution lo, maybe in another gist (just copy paste what happens after `flexget exec`)

**Comment by theboyrossy** [comment]
The link to view my current config is below, https://gist.github.com/theboyrossy/c8f0fcf71101ba6820f6d399477b1195

It's a bit of a mess now as I've tried various things, commented things out etc.

My Flexget execute gives me the following.
https://gist.github.com/theboyrossy/314b879613e0727d97d7a52318f5a632

**Comment by theboyrossy** [comment]
So I have again completely wiped my SD card and reinstalled OSMC and Flexget to try and get to the bottom of my issues.

After reinstalling flexget and following the fixes in recent posts on this site I tested my very simple Showrss feed with a single show for now, Arrow.  
https://gist.github.com/theboyrossy/9bdfa53b1c29ca35d9a1980948734087

This script fails with 
https://gist.github.com/theboyrossy/07dd43ceea780e9485890524780abe6e

As you can see above, I have made very few changes to the script and my RSS feed is very simple. This is a fresh install so I have no idea what the issue could be. As I keep getting errors during the moving file stage could there be some issue moving the files to the external drive??? I'm at a loss.

**Comment by alombarte** [comment]
Looks like the sort-files task is using the "downloaded" status and decides not to sort. I'll see if I can reproduce it and see why this is happening .

**Comment by alombarte** [comment]
Hi @theboyrossy, I've fixed your error in the commit https://github.com/alombarte/raspberry-osmc-automated/commit/18fbc41a04718e45231077c7494573e3b0a5c39c.

Let me know if that works for you and feel free to close the task if so.

**Comment by bernardomaciel** [comment]
I was having the same issue. Your commit fixed it. Also, having the default `config.yml` not to rename the downloaded files was a nice change 👍 

@alombarte is there a way to grab the shows from showRSS automatically, to fill the `myseriesgroup` entry of `config.yml`? I mean, other than copying and pasting...

**Comment by alombarte** [comment]
I don't recall seeing a feed in showRSS returning your list. So I believe the solution would be a script that downloads the html in http://showrss.info/timeline (a `curl` passing data for the auth) and then pipe the output to an xmllint or do a regexp that retrieves the html containing the list.

I haven't really thought a way of doing it since I only have in this list 2 problematic shows out of 25.

**Comment by bernardomaciel** [comment]
OK, thanks. Seems more involved than beneficial.

**Comment by theboyrossy** [comment]
I've had this running for about 10 days now and all seems to be working now :0

Thank you!

**Comment by alombarte** [comment]
Glad to hear! Thanks for reporting back.
